### DISCUSSION TITLE
thunderbird: cannot open email links in firefox

### DISCUSSION THREAD

**Discussion by davidlachambre** [netblue30/firejail#4658] [Q&A]
I'm on Manjaro KDE. Latest everything.

I auto-firejailed everything.

Both firejailed Thunderbird and Firefox work seperately without any problem.

I'm having the following issue.

If I try to open a link in Thunderbird, it prompts me to choose an application to open the link, which suggests it cannot access my default browser. I have both Firefox and Firefox-developer-edition installed (the later is my default by I also tryed using Firefox as my default). I tried pointing to the executable but it did not work. I tried with closed and opened instances of each browser before clicking a link but it did not work.

If I try to run firejail --profile=/etc/firejail/thunderbird.profile firefox

I get the following errors:

GUI: Your Firefox profile cannot be loaded. It may be missing or inaccessible.
Terminal: Failed to load module "appmenu-gtk-module" (was solved by installing it, issue remains)

Relates to:

* #6462

**Comment by topimiettinen** [comment]
Firejail profiles are usually written to construct rather minimal sandboxes, which only allow the specific app to be started and the app can't start further apps. For example, using both TB and FF at same time would need a profile which is combined from both profiles and thus weaker sandbox than just for TB (or FF) only. Implementing #4557 could solve this.

**Comment by rusty-snake** [comment]
The two important issues you need to read about opening links: #3919 and #3919

**Comment by davidlachambre** [comment]
Everything I had read so far about this seemed to indicate that, out of the box, TB should be able to open a link in FF if it is already opened. Is that no longer the case?

**Comment by davidlachambre** [comment]
Adding "nowhitelist ${HOME}/.mozilla/firefox/profiles.ini" in thunderbird.local has it working with Firefox now. Running firejail --profile=thunderbird /usr/bin/firefox https://github.com/ opens a new tab in already running firejailed Firefox. However, using "firejail --profile=thunderbird /usr/bin/firefox-developer-edition https://github.com/" opens a new window in FF dev edition with a different profile (no bookmarks or extensions). What do I need to change/add in order to have the same behavior (open new tab in the same profile) with firefox-developer-edtion as well?
